File: reference/haskell/cli/testMain.hs

module Main where

--------------------------------------------------------------------------------

import Slot ( SlotIdx(..) , DataSource(..) , Seed(..) )
import DataSet
import Sampling 

--------------------------------------------------------------------------------

smallDataSetCfg :: DataSetCfg
smallDataSetCfg = MkDataSetCfg 
  { _maxDepth      = 16
  , _maxLog2NSlots = 5
  , _nSlots    = 5
  , _cellSize  = 128
  , _blockSize = 1024
  , _nCells    = 64
  , _nSamples  = 10
  , _dataSrc   = FakeData (Seed 12345)
  }

bigDataSetCfg :: DataSetCfg
bigDataSetCfg = MkDataSetCfg 
  { _maxDepth      = 32
  , _maxLog2NSlots = 8
  , _nSlots    = 13
  , _cellSize  = 2048
  , _blockSize = 65536
  , _nCells    = 512
  , _nSamples  = 5
  , _dataSrc   = FakeData (Seed 666)
  }

--------------------------------------------------------------------------------

main :: IO ()
main = do
  let dsetCfg = smallDataSetCfg
  let slotIdx = SlotIdx 3
  let entropy = 1234567 :: Entropy
  circomMainComponent dsetCfg                 "./json/slot_main.circom" 
  samplingTest        dsetCfg slotIdx entropy "./json/input_example.json"

File: reference/nim/proof_input/src/testmain.nim

import sugar
import std/sequtils

import constantine/math/arithmetic

import poseidon2/types
import poseidon2/merkle

import types
import blocks
import slot
import dataset
import sample
import merkle
import gen_input
import json

#-------------------------------------------------------------------------------

proc testMerkleProofs*( input: seq[F] ) = 
  let tree = merkleTree(input)
  let root = merkleRoot(input)
  assert( bool(root == treeRoot(tree)) )

  let n  = input.len
  var ok = true
  var oks : seq[bool] = newSeq[bool]( n )
  for i in 0..<n: 
    let proof = merkleProof(tree, i)
    let b = checkMerkleProof(root, proof)
    oks[i] = b
    ok     = ok and b
  let prefix = ("testing Merkle proofs for an input of size " & $n & " ... ")
  if ok:
    echo (prefix & "OK.")
  else:
    echo (prefix & "FAILED!")
    echo oks

proc testAllMerkleProofs*( N: int ) = 
  for k in 1..N:
    let input = collect( newSeq , (for i in 1..k: toF(100+i) ))
    testMerkleProofs( input )

#-------------------------------------------------------------------------------

when isMainModule:
  # testAllMerkleProofs(20)

  let slotIdx  = 3
  let fakedata = DataSource(kind: FakeData, seed: 12345)
  let globcfg  = GlobalConfig( maxDepth: 16, maxLog2NSlots: 5, cellSize: 128, blockSize: 4096)  
  let dsetcfg  = DataSetConfig( nCells: 256, nSlots: 5, nSamples: 10, dataSrc: fakedata)                      
  let entropy  = toF( 1234567 )
  let prfInput = generateProofInput(globcfg, dsetcfg, slotIdx, entropy)
  exportProofInput( "json/foo.json" , prfInput )

